packages/core/src/services/permission/type.ts

```diff
@@ -0,0 +1,125 @@
+/**
+ * Copyright 2023-present DreamNum Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type { Observable } from 'rxjs';
+import { createIdentifier } from '@wendellhu/redi';
+import type { Nullable } from '../../common/type-utils';
+import { LifecycleStages, runOnLifecycle } from '../lifecycle/lifecycle';
+import type { IRange } from '../../types/interfaces';
+
+export enum PermissionStatus {
+    INIT = 'init',
+    FETCHING = 'fetching',
+    DONE = 'done',
+}
+
+export enum PermissionType {
+    WORK_BOOK = 'WORK_BOOK',
+    WORK_SHEET = 'WORK_SHEET',
+    SHEET_RANGE = 'SHEET_RANGE',
+}
+
+export enum UnitPermissionType {
+    Edit = 'Edit',
+    View = 'View',
+    Share = 'Share',
+    Duplicate = 'Duplicate',
+    Export = 'Export',
+    Comment = 'Comment',
+    Print = 'Print',
+    Copy = 'Copy',
+    ManageCollaborator = 'ManageCollaborator',
+    MoveSheet = 'MoveSheet',
+    DeleteSheet = 'DeleteSheet',
+    HideSheet = 'HideSheet',
+    RenameSheet = 'RenameSheet',
+    CreateSheet = 'CreateSheet',
+    CopySheet = 'CopySheet',
+    History = 'History',
+}
+
+export enum SubUnitPermissionType {
+    Edit = 'Edit',
+    Copy = 'Copy',
+    SelectProtectedCells = 'SelectProtectedCells',
+    SelectUnProtectedCells = 'SelectUnProtectedCells',
+    SetCellStyle = 'SetCellStyle',
+    SetCellValue = 'SetCellValue',
+    View = 'View',
+    SetRowStyle = 'SetRowStyle',
+    SetColumnStyle = 'SetColumnStyle',
+    InsertRow = 'InsertRow',
+    InsertColumn = 'InsertColumn',
+    InsertHyperlink = 'InsertHyperlink',
+    DeleteRow = 'DeleteRow',
+    DeleteColumn = 'DeleteColumn',
+    Sort = 'Sort',
+    Filter = 'Filter',
+    PivotTable = 'PivotTable',
+    EditExtraObject = 'EditExtraObject',
+    UnRecognized = 'UnRecognized',
+    ManageCollaborator = 'ManageCollaborator',
+}
+
+export interface IPermissionTypes {
+    rangeTypes?: RangeUnitPermissionType[];
+    worksheetTypes?: SubUnitPermissionType[];
+    workbookTypes?: UnitPermissionType[];
+}
+
+export enum RangeUnitPermissionType {
+    Edit = 'Edit',
+    View = 'View',
+    ManageCollaborator = 'ManageCollaborator',
+}
+export type IUnitPermissionId = `${PermissionType}.${UnitPermissionType}`;
```

Comment:
Same as above. For the permission service, it should treat all permission point as a string, so it could be extensible. https://en.wikipedia.org/wiki/Open%E2%80%93closed_principle